### PR TITLE
Eperez sessions

### PR DESCRIPTION
**Pull request by enriquepablo**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
 from setuptools import setup, find_packages
 
 
-version = '0.1.2'
+version = '0.1.3'
 
-requires = []
+requires = [
+    'pysaml2>=1.0.0',
+    'redis>=2.10.5',
```
Comment (fredrikt) [inline setup.py]: hmm, I don't think we will want to draw in pysaml2 and redis everywhere eduid-common is used? not sure if we've 'scoped' eduid-common to being for web apps...

Comment (enriquepablo) [inline setup.py]: eduid_common is a namespace package, so at any moment we can put eduid_common.session (and/or eduid_common.authn) into its own github repo/pypi egg, removing the dependencies from core eduid_common.

Comment (enriquepablo) [inline setup.py]: As to being scoped for web apps, I'm not treating it that way, I'm assuming that it should contain whatever may be commonly used by the microservices and the pyramid apps.

File: src/eduid_common/session/session.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        :type token: str
+
+        :return: the unsigned key
+        :rtype: str
+        '''
+        #  the slicing is to remove a leading 'a' needed so we have a
+        # valid NCName so pysaml2 doesn't complain when it uses the token as
+        # session id.
+        val = token.strip('"')[1:]
+        sig = HMAC.new(self.secret, val[40:].encode('utf-8'), SHA1).hexdigest()
```
Comment (fredrikt) [inline src/eduid_common/session/session.py]: SHA1 is not recommended for new designs, can we change it to SHA2-256?

Comment (enriquepablo) [inline src/eduid_common/session/session.py]: right away :)

**Pull request merged**